File: Interview150/Ransom-Note.cpp
#include <map>
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        map<char, int> mag;
        // cout << mag.size();
        for (auto &ch : magazine) { 
            mag[ch]++;
        } 
        for (auto &ch : ransomNote){
            if (mag[ch] == 0) return false;
            mag[ch]--;
        }
        return true;
        
    }
};

File: Interview150/Ransom-Note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        mag = {}
        for c in magazine:
            x = mag.get(c,0)
            mag[c] = x+1
        for c in ransomNote:
            x = mag.get(c,0)
            if x==0:
                return False
            mag[c] = x-1
        return True
        